File: Car Pooling.js
var carPooling = function(trips, capacity) {
    var vehicle = [capacity];
    var i = trips[0][1];
    var sum = 0;
    for(var t of trips){
        vehicle[t[1]] =(vehicle[t[1]] || 0) -t[0];
        vehicle[t[2]] =(vehicle[t[2]] || 0) +t[0];
    }
    for(var x of vehicle){
        if(x) sum += x;
        if(sum < 0) return false;
    }
    return true;
};

File: Car Pooling.py
class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        distance = [0]*1002

        for num, start, end in trips:
            distance[start] += num
            distance[end] -= num

        for i  in range(1001):
            distance[i] += distance[i-1]
            
        for i in range(1001):
            if distance[i] > capacity:
                return False

        return True
